tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py

```diff
@@ -59,6 +81,21 @@ def take_photo(self):
         # Wait for thumbnail to appear
         self.wait_for_thumbnail_visible()
 
+    def take_photo_with_timer(self):
+        # Wait for camera to be ready to take a picture
+        controls = self.marionette.find_element(*self._controls_locator)
+        Wait(self.marionette, timeout=20).until(
+            lambda m: controls.get_attribute('data-enabled') == 'true')
+         
+        self.tap_capture()
+        time.sleep(1)
```

Comment:
In fact, this method looks very similar to the take_photo method. Why not use the existing take_photo method and add an extra argument with_timer that defaults to False, but in case you want to use it is doing the checks that you added here?